Include/cpython/code.h

```diff
@@ -72,6 +72,24 @@ typedef struct {
     uint8_t *per_instruction_tools;
 } _PyCoMonitoringData;
 
+#ifdef Py_GIL_DISABLED
+
+/* Each thread specializes a thread-local copy of the bytecode in free-threaded
+ * builds. These copies are stored on the code object in a `_PyCodeArray`. The
+ * first entry in the array always points to the "main" copy of the bytecode
+ * that is stored at the end of the code object.
+ */
+typedef struct {
+    Py_ssize_t size;
+    char *entries[1];
```

Comment:
I started out with this, but it would require including `pycore_code.h`, which would introduce a circular dependency between the two files. That should probably be sorted out in a separate PR. I'm guessing that's also why `co_code_adaptive` is a `char` array and not a `_Py_CODEUNIT` array:

https://github.com/python/cpython/blob/99400930ac1d4e5e10a5ae30f8202d8bc2661e39/Include/cpython/code.h#L140

Reply:
That makes sense. We should probable move `PyCodeObject` to an internal header sometime in the future.